vendor/bat-native-ads/src/bat/ads/internal/resources/contextual/text_classification/text_classification_resource.cc

```diff
@@ -34,30 +49,41 @@ bool TextClassification::IsInitialized() const {
 }
 
 void TextClassification::Load() {
-  AdsClientHelper::Get()->LoadAdsResource(
+  auto weak_ptr = weak_ptr_factory_.GetWeakPtr();
+  AdsClientHelper::Get()->LoadAdsFileResource(
       kResourceId, features::GetTextClassificationResourceVersion(),
-      [=](const bool success, const std::string& json) {
-        text_processing_pipeline_.reset(
-            ml::pipeline::TextProcessing::CreateInstance());
+      [weak_ptr](base::File file) {
+        base::ThreadPool::PostTaskAndReplyWithResult(
+            FROM_HERE, {base::MayBlock()},
+            base::BindOnce(&ReadFileToString, std::move(file)),
+            base::BindOnce(
+                &TextClassification::CreateTextClassificationOnMainThread,
+                weak_ptr));
+      });
+}
 
-        if (!success) {
-          BLOG(1, "Failed to load " << kResourceId
-                                    << " text classification resource");
-          return;
-        }
+void TextClassification::CreateTextClassificationOnMainThread(
```

Comment:
@tmancey @aseren 
Could you guys also consider moving parsing the file on ThreadPool too?
There is no simple option to do it now, because it requires touching `text_processing_pipeline_` on a worker thread (or holding 100mb base::Value until the main thread becomes ready).

Reply:
Great idea! Thanks. Currently it won't give a performance boost as bat ads startup logic is linear and single threaded.
Filed a ticket to do refactoring and optimization: https://github.com/brave/brave-browser/issues/22033